src/libopenrave/kinbody.cpp

```diff
@@ -5459,6 +5459,9 @@ void KinBody::ExtractInfo(KinBodyInfo& info)
     _ResolveInfoIds();
 
     info._id = _id;
+    if(info._id.empty()) {
```

Comment:
@bucketzxm This seems out of place. Like other infos, perhaps you need to do it in UpdateFromInfo

Reply:
if we load a `.dae` file into `env`, and do the following command.  Then we need to set id before creating newkinbody,  or set id when we do extract info 
```
emptyEnv.UpdateFromInfo(env.ExtractInfo())
```